### PR TITLE
Handle /nodes/families json requests properly (bsc#926395)

### PR DESCRIPTION
**Pull request by rsalevsky**
Handle json request properly.

### CODE REVIEW

**Comment by aspiers** [comment]
Please make the commit message explain the change, not just refer to a bugzilla entry (which is probably only accessible by SUSE employees).  Also it is best practice to include the full URL to the bugzilla entry in the body of the commit message.  (This is [documented internally](https://etherpad.nue.suse.com/p/cloud-pull-requests) BTW.)

Here's an example of the ideal commit message: https://github.com/crowbar/barclamp-pacemaker/commit/ad8c04e5f6f0daf4c896b8ba54d96745cda3dd5d

**Comment by rsalevsky** [comment]
@aspiers You are to fast. :P Was just a test to see how travis handles the unit tests.

**Comment by aspiers** [comment]
Hah :)

File: crowbar_framework/app/controllers/nodes_controller.rb [hunk 1]
```diff
@@ -219,6 +219,10 @@ def families
         })
       end
     end
+
+    respond_to do |format|
+      format.html { render "families" }
```
Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb]: Why? You can skip the render. Conventional you want to write just ```format.html``` and nothing more in that case.

File: crowbar_framework/spec/controllers/nodes_controller_spec.rb [hunk 2]
```diff
@@ -154,6 +154,11 @@
       get :families
       assigns(:families).keys.should include(node.family.to_s)
     end
+
+    it "as json is not acceptable" do
```
Comment (tboerger) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: IMHO not a really useful spec. We don't really care if it responds to JSON or something different as long as a proper page gets displayed.

**Comment by bkutil** [comment]
@aspiers @tboerger the pull is marked as WIP, so give the guy some space to experiment :) You can always shoot it to pieces with your comments later :)

**Comment by tboerger** [comment]
> @aspiers @tboerger the pull is marked as WIP, so give the guy some space to experiment :) You can always shoot the patch to pieces with your comments later :)

I don't want to shoot his patch in pieces. I just want to give early feedback to improve his stuff :)

File: crowbar_framework/spec/controllers/nodes_controller_spec.rb (hunk 2 above)
Comment (rsalevsky) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: So you mean we can skip this test?

Comment (tboerger) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: I think so. @bkutil what do you say?

Comment (dirkmueller) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: are we really arguing against a regression test here?

Comment (tboerger) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: > I think we're generally preferring TDD

Sure, but useful tests.

Comment (dirkmueller) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: it was a bugreport by QA, it makes sense to add a test that checks that there is not again a 500. plus its a nice learning excercise on how to write specs..

**Comment by aspiers** [comment]
@bkutil It wasn't marked WIP when I reviewed it ;-)  But yes like @tboerger I'm just trying to be helpful ;-)

File: crowbar_framework/spec/controllers/nodes_controller_spec.rb (hunk 2 above)
Comment (tboerger) [inline crowbar_framework/spec/controllers/nodes_controller_spec.rb]: This spec doesn't work anyway. Travis fails because of that.

**Comment by bkutil** [comment]
@rsalevsky - I'm not 100% sure, but I think @vmoravec 's bug report was about the action not being able to handle requests for `JSON` If so, a proper fix would be then:

1. add `render json: @families` to the controller
2. add a trivial spec checking that response is success and it contains something meaningful.

**Comment by tboerger** [comment]
@bkutil The report was that he get a 500 error page when he tries to access it with JSON format. Instead of a 500 it should simply print a 404 error and nothing more. IMHO we don't need the family stuff in JSON format.

**Comment by vuntz** [comment]
I guess the question is whether this is part of the API or not. If it is, then json should be an option. If not, then 404 is fine.

**Comment by tboerger** [comment]
> I guess the question is whether this is part of the API or not. If it is, then json should be an option. If not, then 404 is fine.

Currently we are using barely the HTML output, but not the JSON output. So from my side a 404 should be fine.

**Comment by rsalevsky** [comment]
Ready for review

**Comment by tboerger** [comment]
:+1:

File: crowbar_framework/app/controllers/nodes_controller.rb [hunk 3]
```diff
@@ -219,6 +219,10 @@ def families
         })
       end
     end
+
+    respond_to do |format|
+      format.html
+    end
```
Comment (vuntz) [inline crowbar_framework/app/controllers/nodes_controller.rb]: Can we get a comment at the beginning of the method to say that this is a UI-only method, not an API method?

**Comment by dirkmueller** [comment]
+1

**Pull request merged**